core/board.py

```diff
@@ -0,0 +1,91 @@
+from typing import List, Any
+
+
+class Board:
+    def __init__(self):
+        self.size = 3
+        self.grid = list(range(self.size * self.size))
+        # Hard-coded... Not scalable if we move to 4x4, 5x5, etc.
+        self.win_combinations = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]]
+
+    def get_available_spots(self):
+        available_spots = [str(s) for s in self.grid if s != "X" and s != "O"]
+        return available_spots
+
+    def is_empty(self):
+        return len(self.get_available_spots()) == 9
+
+    def reset(self):
+        self.grid = list(range(9))
+
+    def set_spot(self, spot: int, symbol: str) -> None:
+        self.grid[spot] = symbol
+
+    def is_win(self):
+        all_combinations: List[list] = []
+
+        all_rows: List[list] = self.get_all_rows()
+        for row in all_rows:
+            all_combinations.append(row)
+
+        all_columns: List[list] = self.get_all_columns()
+        for column in all_columns:
+            all_combinations.append(column)
+
+        all_diagonals: List[list] = [self.get_first_diagonal(), self.get_second_diagonal()]
```

Comment:
It's not clear what first and second diagonal are. However, naming such as `top_left_to_bottom_right_diagonal` and `bottom_left_to_top_right_diagonal` would be clear.